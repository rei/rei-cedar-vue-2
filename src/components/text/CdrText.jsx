import style from './styles/CdrText.scss';

/**
 * Cedar 2 component for text styles
 * Accepts typography modifiers.
 * @version 0.0.1
 * @author [REI Software Engineering](https://rei.github.io/rei-cedar/)
 */
export default {
  name: 'CdrText',
  props: {
    /** Any valid HTML tag */
    tag: {
      type: String,
      default: 'p',
    },
  },
  data() {
    return {
      s: style, // named `s` not `style` so that modifierClass works for text utils
    };
  },
  computed: {
    baseClass() {
      return 'cdr-text';
    },
  },
  render() {
    const Component = this.tag;
    return (<Component class={this.s[this.baseClass]}>
      {this.$slots.default}
    </Component>);
  },
};
