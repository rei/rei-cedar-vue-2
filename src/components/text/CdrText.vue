<template>
  <!-- disable lint errors on line length in template -->
  <!-- eslint-disable max-len -->
  <component
    :is="tag"
    :class="[modifierClass]"
  >
    <slot/>
  </component>
</template>

<script>
import modifier from 'mixinsdir/modifier';

/**
 * Cedar 2 component for text styles
 * Accepts typography modifiers.
 * @version 0.0.1
 * @author [REI Software Engineering](https://rei.github.io/rei-cedar/)
 */
export default {
  name: 'CdrText',
  mixins: [modifier],
  props: {
    /** Any valid HTML tag */
    tag: {
      type: String,
      default: 'p',
    },
  },
  computed: {
    baseClass() {
      return 'cdr-text';
    },
  },
};
</script>
