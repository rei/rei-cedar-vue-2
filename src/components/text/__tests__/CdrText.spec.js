import { mount } from '@vue/test-utils';
import CdrText from 'componentdir/text/CdrText';

describe('CdrText', () => {
  test('renders correctly', () => {
    const wrapper = mount(CdrText);
    expect(wrapper.element).toMatchSnapshot();
  });
  
  it('renders as h1 correctly', () => {
    const wrapper = mount(CdrText, {
      propsData: {
        tag: 'h1',
      },
    });
    expect(wrapper.element).toMatchSnapshot();
  });
});