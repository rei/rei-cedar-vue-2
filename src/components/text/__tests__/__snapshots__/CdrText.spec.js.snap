// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CdrText renders as h1 correctly 1`] = `
<h1
  class="cdr-text"
/>
`;

exports[`CdrText renders correctly 1`] = `
<p
  class="cdr-text"
/>
`;
